src/System.CommandLine.Tests/Invocation/CancelOnProcessTerminationTests.cs

```diff
@@ -91,6 +91,169 @@ public async Task CancelOnProcessTermination_cancels_on_process_termination(int
+
+                    try
+                    {
+                        context.Console.WriteLine(ChildProcessWaiting);
+                        await Task.Delay(int.MaxValue, cancellationToken);
+                        context.ExitCode = 1;
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        // For Process.Exit handling the event must remain blocked as long as the
+                        // command is executed.
+                        // We are currently blocking that event because CancellationTokenSource.Cancel
+                        // is called from the event handler.
+                        // We'll do an async Yield now. This means the Cancel call will return
+                        // and we're no longer actively blocking the event.
+                        // The event handler is responsible to continue blocking until the command
+                        // has finished executing. If it doesn't we won't get the CancelledExitCode.
+                        await Task.Yield();
+
+                        context.ExitCode = CancelledExitCode;
+                    }
+
+                    // This is an example of bad pattern and reason why we need a timeout on termination processing
+                    await Task.Delay(TimeSpan.FromMilliseconds(2000));
+
+                    // Execution should newer get here as termination processing has a timeout of 100ms
+                    context.ExitCode = 123;
+                });
+
+                return new CommandLineBuilder(new RootCommand
+                       {
+                           command
+                       })
+                        // Unfortunately we cannot use test parameter here - RemoteExecutor currently doesn't capture the closure
+                       .CancelOnProcessTermination(TimeSpan.FromMilliseconds(100))
+                       .Build()
+                       .InvokeAsync("the-command");
+            };
+
+            using RemoteExecution program = RemoteExecutor.Execute(childProgram, psi: new ProcessStartInfo { RedirectStandardOutput = true });
+
+            Process process = program.Process;
+
+            // Wait for the child to be in the command handler.
+            string childState = await process.StandardOutput.ReadLineAsync();
+            childState.Should().Be(ChildProcessWaiting);
+
+            // Request termination
+            kill(process.Id, signo).Should().Be(0);
+
+            // Verify the process terminates timely
+            bool processExited = process.WaitForExit(10000);
+            if (!processExited)
+            {
+                process.Kill();
+                process.WaitForExit();
+            }
+            processExited.Should().Be(true);
+
+            // Verify the process exit code
+            process.ExitCode.Should().Be(timeOutMs.HasValue ? ForceTerminationCode : CancelledExitCode);
```

Comment:
`timeOutMs` always has a value in this `Theory`. I'd suggest removing the branching logic or, if the two theories' implementations are substantially similar, hoist the exit codes into the `InlineData` definitions and only have one `Theory`.

Reply:
Unfortunately it's not possible to have that parameter in `InlineData` as `RemoteExecutor` doesn't capture the closure variables - so parameters would not get propagated (hence the reason for 2 test methods).

I'll remove that parameter to make it less confusing